redfish/provider/resource_redfish_user_account.go

```diff
@@ -1,5 +1,412 @@
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/stmcginnis/gofish"
+	"github.com/stmcginnis/gofish/redfish"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource = &UserAccountResource{}
+)
+
+// NewUserAccountResource is a helper function to simplify the provider implementation.
+func NewUserAccountResource() resource.Resource {
+	return &UserAccountResource{}
+}
+
+// UserAccountResource is the resource implementation.
+type UserAccountResource struct {
+	p *redfishProvider
+}
+
+// Configure implements resource.ResourceWithConfigure
+func (r *UserAccountResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	r.p = req.ProviderData.(*redfishProvider)
+}
+
+// Metadata returns the resource type name.
+func (r *UserAccountResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "UserAccount"
+}
+
+// Schema defines the schema for the resource.
+func (r *UserAccountResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "This Terraform resource is used to manage user entity of the iDRAC Server. We can create, read, modify and delete an existing user using this resource.",
+		Description:         "This Terraform resource is used to manage user entity of the iDRAC Server. We can create, read, modify and delete an existing user using this resource.",
+		Version:             1,
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "The ID of the user. Cannot be updated.",
+				Description:         "The ID of the user. Cannot be updated.",
+				Computed:            true,
+			},
+			"user_id": schema.StringAttribute{
+				MarkdownDescription: "The ID of the user. Cannot be updated.",
+				Description:         "The ID of the user. Cannot be updated.",
+				Optional:            true,
+				Computed:            true,
+			},
+			"username": schema.StringAttribute{
+				MarkdownDescription: "The name of the user",
+				Description:         "The name of the user",
+				Required:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(1, 16),
+				},
+			},
+			"password": schema.StringAttribute{
+				MarkdownDescription: "Password of the user",
+				Description:         "Password of the user",
+				Required:            true,
+				Sensitive:           true,
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(4, 40),
+				},
+			},
+			"redfish_server": schema.SingleNestedAttribute{
+				MarkdownDescription: "Redfish Server",
+				Description:         "Redfish Server",
+				Required:            true,
+				Attributes:          RedfishServerSchema(),
+			},
+			"role_id": schema.StringAttribute{
```

Comment:
Probably better to add "None" in the Default field, rt?

Reply:
Updated